src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,543 @@
+        /// <param name="task">The task for the log entry</param>
+        /// <param name="opcode">The operation for the log entry</param>
+        /// <param name="level">The logging level.</param>
+        /// <param name="keyword">The keyword(s) for the event.</param>
+        /// <param name="args">The payload for the log message</param>
+        public void Log(PSEventId eventId, PSChannel channel, PSTask task, PSOpcode opcode, PSLevel level, PSKeyword keyword, params object[] args)
+        {
+            if (ShouldLog(level, keyword, channel))
+            {
+                int threadId = Thread.CurrentThread.ManagedThreadId;
+
+                StringBuilder sb = MessageBuilder;
+                sb.Clear();
+
+                // add the message preamble
+                sb.AppendFormat(CultureInfo.InvariantCulture,
+                                "({0:X}:{1:X}:{2:X}) [{3:G}:{4:G}.{5:G}.{6:G}] ",
+                                _commitId, threadId, channel, eventId, task, opcode, level);
+
+                // add the message
+                GetEventMessage(sb, eventId, args);
+
+                NativeMethods.SysLogPriority priority;
+                if ((int) level <= _levels.Length)
+                {
+                    priority = _levels[(int)level];
+                }
+                else
+                {
+                    priority = NativeMethods.SysLogPriority.Info;
+                }
+                // log it.
+                NativeMethods.SysLog(priority, sb.ToString());
+            }
+        }
+
+        #endregion logging
+    }
+
+    internal enum LogLevel : uint
+    {
+        Always = 0,
+        Critical = 1,
+        Error = 2,
+        Warning = 3,
+        Information = 4,
+        Verbose = 5
+    }
+
+    internal static class NativeMethods
+    {
+        /// <summary>
+        /// Write a message to the system logger, which in turn writes the message to the system console, log files, etc.
+        /// See man 3 syslog for more info
+        /// </summary>
+        /// <param name="priority">
+        /// The OR of a priority and facility in the SysLogPriority enum indicating the the priority and facility of the log entry
+        /// </param>
+        /// <param name="message">The message to put in the log entry</param>
+        [DllImport("psl-native", CharSet = CharSet.Ansi, EntryPoint = "Native_SysLog")]
+        internal static extern void SysLog(SysLogPriority priority, string message);
+
+        [DllImport("psl-native", CharSet = CharSet.Ansi, EntryPoint = "Native_OpenLog")]
+        internal static extern void OpenLog(IntPtr ident, SysLogPriority facility);
+
+        [DllImport("psl-native", EntryPoint = "Native_CloseLog")]
+        internal static extern void CloseLog();
+
+        [Flags]
+        internal enum SysLogPriority : uint
+        {
+            // Priorities
+            /// <summary>
+            /// system is unusable
```

Comment:
Please reformat public XML comments in the enum - start with capital and add final dot.

Reply:
This is not a public type.